main.py

```diff
@@ -1,14 +1,84 @@
+import argparse
 from config import SEMANTIC_SCHOLAR_CITATIONS_DATASET_NAME
 from data_retrival.csv_handler import convert_citations_json_to_csv
 from data_retrival.download_dataset_handler import DatasetHandler
 
 ### --> In progress <-- ###
 # Playground #
 
+def make_a_parser():
+    argParser = argparse.ArgumentParser(description="Download and process Semantic Scholar citations dataset")
+
+    subparsers = argParser.add_subparsers(dest="command", help="Manage the download and processing of the Semantic Scholar citations dataset")
+
+    # DOWNLOADS
+    download_links_parser = subparsers.add_parser("get-links", help="Download links for batches download")
+
+    download_batches_parser = subparsers.add_parser("get-batches", help="Download batches of citations dataset")
+    download_batches_parser.add_argument(
+        "--all", action="store_true", help="Download all batches of citations dataset, otherwise download only next batch"
+    )
+    # PROCESSING
+    process_batches_parser = subparsers.add_parser("process-batches", help="Process downloaded batches of citations dataset from json to csv")
+    process_batches_parser.add_argument(
+        "--all", action="store_true", help="Process all downloaded batches to csv of citations dataset, otherwise process only next batch"
+    )
+
+    # GRAPH BUILDING
+    build_on_graph_parser = subparsers.add_parser("build-on-graph", help="Build on graph from processed csv files")
```

Comment:
Move it to args.py file, in the same dir as main, just for readbility

Reply:
It looks like line formatting is missing :) Install pre-commit tool, please see our Readme file